daemon/networkdriver/bridge/driver.go

```diff
@@ -140,13 +141,36 @@ func InitDriver(job *engine.Job) engine.Status {
 		// Bridge exists already. Getting info...
 		// validate that the bridge ip matches the ip specified by BridgeIP
 		if bridgeIP != "" {
-			networkv4 = addrv4.(*net.IPNet)
 			bip, _, err := net.ParseCIDR(bridgeIP)
+			networkv4 = addrv4.(*net.IPNet)
 			if err != nil {
 				return job.Error(err)
 			}
 			if !networkv4.IP.Equal(bip) {
-				return job.Errorf("bridge ip (%s) does not match existing bridge configuration %s", networkv4.IP, bip)
+				// Configured bridge IP (by --bip="xxxx/x") is not the same with the real bridge IP.
+				// Add a new bridge IP here and remove the old one.
```

Comment:
I think we should at least add a log entry that the current bridge IP was different than the flag so that there is some "audit" for why/when the bridge device IP changed.  Obviously the passing of the flag should make the user aware that they are requesting it be changed, but it would probably still be good to log.